supervisor/homeassistant.py

```diff
@@ -559,12 +563,21 @@ async def make_request(
 
     async def check_api_state(self) -> bool:
         """Return True if Home Assistant up and running."""
+        # Check if port is up
+        if not await self.sys_run_in_executor(
+            check_port, self.ip_address, self.api_port
+        ):
+            return False
+
+        # Check if API is up
         with suppress(HomeAssistantAPIError):
-            async with self.make_request("get", "api/") as resp:
+            async with self.make_request("get", "api/config") as resp:
                 if resp.status in (200, 201):
-                    return True
-                status = resp.status
-            _LOGGER.warning("Home Assistant API config mismatch: %s", status)
+                    data = await resp.json()
+                    if data.get("state", "RUNNING") == "RUNNING":
```

Comment:
Why would we wait on running for the rollback check? The frontend would be up already the user would be able to configure or fix whatever problem there could be, so why not just check if the frontend is up?

Reply:
Let's merge P2 and we can start to play with that. You can after P2 start to add new integration or remove integration etc while HA is startup? We can later split up the check for block startup and API proxy 👍